### ISSUE TITLE
Better Library Search/"All Designs" Tab + Favourites

### ISSUE THREAD

**Issue by Gibsonion** [soflyy/oxygen-bugs-and-features#340]
Now the design library is getting quite large it would be useful to have a better way of navigating through the available parts. 

If I'm searching for a Hero Section/content area/footer it's very time consuming switching between each design set, to find what I'm looking for.

A new tab that shows all the parts separated by category would make this process much more streamlined.

Or some way of marking a custom shortlist of favourites for easy recall and navigation. Especially if it could be imported/exported between builds.

**Uses:**
Massive time saver

**Comment by INTELLIPLANS** [comment]
Agreed,  Will this also impact the CSS files associated with all of the respective sections/elements?

**Comment by Gibsonion** [comment]
If you mean will it prevent needing to load lots of unwanted CSS when comparing the different components, hopefully yes.

**Comment by wplit** [comment]
I agree there could be a better way. 

The problem I see with organising by category though is that it mean your site would need to connect with all the sites holding the design sets in one go. Currently, it doesn't connect with any them until you actually select which design set, then it connects to that site and fetches all the elements/categories. That's why there's that little delay as they load up. If you had organise by category, in order to know what categories to show, it would first need to connect to all of the sites and pull in all of the design sets. It would be pretty slow depending on how many sites you had.

**Comment by Gibsonion** [comment]
Surely then it should be very simple if that's how libraries are made, just make a site with all the components into a new library, it might be a little slower to initially load, but a hell of a lot quicker than having to wait every time you switch libraries.
